### PR TITLE
Add Reverse registrar

### LINKED ISSUE #114: ENS (reverse registrar) for displaying addresses
### Challenge description:

if an address is associated with an ENS name, show that, rather than the address itself.

See https://docs.ens.domains/contract-api-reference/reverseregistrar

This way we can show stuff like "adex-recovery.ens" rather than "0x...

This can be achieved by creating a address component that will automatically look it up on the reverse registrar.

### Submission requirements:

* The solution is presented in the form of a pull request to this repo, and adheres to React/Redux best practices

### Judging criteria:

* First to submit a working submission that meets our quality standards/criteria

### Submission deadline:

August 19

### Bounty:

250 DAI

### PR DESCRIPTION
**Pull request by iamonuwa**
Fixes #114

### CODE REVIEW

**Comment by IvoPaunov** [comment]
- remove semicolon to keep the code style consistent
- make the ens check with [action](https://github.com/AdExNetwork/adex-dapp/tree/master/src/actions) and [service](https://github.com/AdExNetwork/adex-dapp/tree/master/src/services) - Its small piece of code but with ENS service you can add new functionalities easier in the future
- don't commit the package-lock
- use the component where needed

File: src/components/common/ens_address/ens_address.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  		if (isEthAddress(address)) {
+  			this.setState({
+  				isSearching: true
+  			});
+
+  			console.log(address.indexOf(""));
+
+  			await provider.lookupAddress(address).then(async ensName => {
+  				if (ensName !== null) {
+  					const resolvedAddress = await provider.resolveName(ensName);
```
Comment (IvoPaunov) [inline src/components/common/ens_address/ens_address.js]: use try/catch if needed. I'm not sure but this kind of actions may crash.

File: src/components/common/ens_address/ens_address.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  					if (resolvedAddress === address) {
+  						this.setState({
+  							isSearching: false
+  						});
+  						updateENSResolution({ ensName, address });
+  					}
+  				} else {
+  					this.setState({
+  						error: "Address has no ens name associated to it.",
```
Comment (IvoPaunov) [inline src/components/common/ens_address/ens_address.js]: use [adex-translations](https://github.com/AdExNetwork/adex-translations)
add the string to `en-US.json` with pull request.

use `import Translate from 'components/translate/Translate'` `(Translate(withStyles(styles)(Connect))`)

File: src/components/common/ens_address/styles.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+export const styles = theme => ({
```
Comment (IvoPaunov) [inline src/components/common/ens_address/styles.js]: do not use random sizes, use `theme.spacing.unit` etc.

**Comment by iamonuwa** [comment]
> use the component where needed

Can you list where needed? I can't login.

**Pull request closed without merging**